Datasources: Add documentation around secure socks proxy (#66609)
grafana/grafana: docs/sources/setup-grafana/configure-grafana/proxy.md

---
aliases:
  - ../features/datasources/proxy/
description: Proxy datasource connections through a secure socks proxy.
keywords:
  - proxy
  - guide
title: Proxying data source connections
weight: 1110

> ---

Reviewer: @stephaniehingtgen , general comment - under the configure-grafana directory, you should create a `proxy` subdirectory and within that directory, add an `index.md` file. The index.md file contains the content.

> # Proxying data source connections

Reviewer: @stephaniehingtgen , we typically avoid gerunds in topic titles. Can this be "Proxy a data source connections"? Or "Configure a data source connection proxy"?

Author: For sure! Changed it to `Configure a data source connection proxy`

Grafana provides support for proxying data source connections through a Secure Socks5 Tunnel. This enables you to securely connect to data sources hosted in a different network than Grafana.

> In order to make use of this functionality, you will need to deploy a socks5 proxy server that supports TLS on a machine exposed to the public internet within the same network as your data source. From there, Grafana will set up a mutually trusted connection from Grafana to the Proxy, and then the Proxy can proxy the Grafana connection to your private server, without exposing your data sources to the public internet.

Reviewer: ```suggestion
To make use of this functionality, you need to deploy a socks5 proxy server that supports TLS on a machine exposed to the public internet within the same network as your data source. From there, Grafana establishes a mutually trusted connection from Grafana to the Proxy. Then the Proxy can proxy the Grafana connection to your private server without exposing your data sources to the public internet.
```

## Known limitations

- Only one socks5 proxy can be configured per Grafana instance
- All built-in core data sources are compatible, but not all external data sources are. See the [private datasource connect docs](https://grafana.com/docs/grafana-cloud/data-configuration/configure-private-datasource-connect/#known-limitations) for an updated list of supported data sources.

> ## Configure Grafana to use the proxy

Reviewer: @stephaniehingtgen , this is a stacked heading, which is one heading followed immediately by another heading. We want to avoid this kind of structure. I recommend removing line 23.

## Before you begin

> Deploy a socks proxy server that:

Reviewer: ```suggestion
To follow this guide, you must first deploy a socks proxy server that:
```

Reviewer: Are there any other before you begin items? Any special permissions or access that the user should have?

- Supports TLS
- Is exposed to the public internet
- Is hosted within the same network as your data source

> ### Configure Grafana to proxy connections through a socks server

Author: Definitely! There is a pre-req of deploying the proxy server, and then configuring grafana
